Hamcore.c: handle memory allocation error

chipitsine opened the pull request.

davidebeatrici commented on Hamcore.c:
```diff
@@ -277,9 +281,21 @@ bool HamcoreBuild(const char *dst_path, const char *base_path, const char **src_
 
 		const size_t path_size = strlen(relative_path) + 1;
 		file->Path = malloc(path_size);
+		if (!file->Path)
+		{
+			free(buffer);
+			return false;
+		}
```
```suggestion
		if (!file->Path)
		{
			free(compressed_files);
			free(buffer);
			return false;
		}
```

davidebeatrici commented on Hamcore.c:
```diff
@@ -277,9 +281,21 @@ bool HamcoreBuild(const char *dst_path, const char *base_path, const char **src_
 		memcpy(file->Path, relative_path, path_size);
 
 		compressed_file->Data = malloc(file->Size);
+		if (!compressed_file->Data)
+		{
+			free(buffer);
+			return false;
+		}
```
```suggestion
		if (!compressed_file->Data)
		{
			free(compressed_files);
			free(buffer);
			return false;
		}
```

davidebeatrici commented:
Could you squash the commits, please?

chipitsine commented:
sure, will do soon

chipitsine committed:
Hamcore.c: handle memory allocation error

Hamcore.c       284     warn    V575 The potential null pointer is passed into 'memcpy' function. Inspect the first argument. Check lines: 284, 283.
Hamcore.c       287     warn    V575 The potential null pointer is passed into 'memcpy' function. Inspect the first argument. Check lines: 287, 286.
Hamcore.c       235     warn    V522 There might be dereferencing of a potential null pointer 'compressed_files'. Check lines: 235, 215.
Hamcore.c       236     warn    V522 There might be dereferencing of a potential null pointer 'compressed_file'.

chipitsine commented:
squashed
